app/src/main/java/org/thoughtcrime/securesms/database/MmsDatabase.kt

```diff
@@ -1403,22 +1331,17 @@ class MmsDatabase(context: Context, databaseHelper: SQLCipherOpenHelper) : Messa
             val contacts: List<Contact?> = getSharedContacts(
                 cursor, attachments
             )
-            val contactAttachments =
-                contacts.map { obj: Contact? -> obj!!.avatarAttachment }
-                    .filter { a: Attachment? -> a != null }
-                    .toSet()
+            val contactAttachments: Set<Attachment?> =
+                contacts.mapNotNull { obj: Contact? -> obj!!.avatarAttachment }.toSet()
```

Comment:
Hmm, I wonder if a `filterNotNull` makes sense here to get rid of the use of `!!` for `obj` and below with `lp` and `o`

Reply:
that way it could become `contacts.filterNotNull().mapNotNull(Contact::avatarAttachment).toSet()` or something (I haven't thought it completely through yet)

Reply:
sure.

`contacts.mapNotNull { it?.avatarAttachment }.toSet()`

just wasn't sure it'd change the semantic. if it was supposed to throw or something. but probably not, no.